pkg/speccheck/builtin/process_secret.go

```diff
@@ -11,6 +11,10 @@ import (
 type secret struct {
 }
 
+func (secret) BeforeCreate(ctx *speccheck.Context, spec *unstructured.Unstructured) (runtime.Object /*updatedSpec*/, error) {
```

Comment:
All processors except for the ServiceInstance and Deployment ones just return the spec verbatim.